tutorial/tests/test_basic_datatypes.py

```diff
@@ -467,5 +472,47 @@ def reference_string_split_lines(my_string: str) -> list[str]:
 
 def test_string_split_lines(function_to_test):
     my_string = "\n".join(STRINGS)
-    assert function_to_test(my_string) == STRINGS
     assert function_to_test(my_string) == reference_string_split_lines(my_string)
+
+
+INT_SETS = [set(), {1, 2, 3}, {3, -1, 0, 4, 42, 1002}]
+
+
+def reference_sets_of_even_and_odd(my_set: set[int]) -> tuple(set[int]):
+    even_set = my_set.copy()
+    odd_set = my_set.copy()
+    for n in my_set:
+        if n % 2 == 0:
+            odd_set.remove(n)
+        else:
+            even_set.remove(n)
+    return even_set, odd_set
+
+
+@pytest.mark.parametrize("my_set", INT_SETS)
+def test_sets_of_even_and_odd(my_set, function_to_test):
+    sol_set = my_set.copy()
+    ref_set = my_set.copy()
```

Comment:
Why do we need to copy here? We are already copying inside the reference solution?

Reply:
Technically i only need two clean copies of myset, so one copy would suffice. But if feels more natural to make two new copies and feed them to each of the solution/reference. It's only important that i dont use the same set for function_to_test and the reference solution becuase functoin_to_test might change the set. then my_set stays untouched and we get clean solution functions for both functions. This is then compatible if we for some reason change the reference solution or run a function multiple times for different tests.